File: test/test_transform_iter.cpp
#include <stdio.h>
#include "TRTCContext.h"
#include "DVVector.h"
#include "fake_vectors/DVTransform.h"
#include "transform.h"

int main()
{
	

	Functor square_root{ {}, { "x" }, "        return sqrtf(x);\n" };

	float hvalues[8] = { 1.0f, 4.0f, 9.0f, 16.0f };
	DVVector dvalues("float", 4, hvalues);

	float houtput[4];
	DVVector doutput("float", 4);

	DVTransform dtrans(dvalues, "float", square_root);

	TRTC_Transform(dtrans, doutput, Functor("Negate"));
	doutput.to_host(houtput);
	printf("%f %f %f %f\n", houtput[0], houtput[1], houtput[2], houtput[3]);

	return 0;
}

File: CSharp/test/test_transform_iter/test_transform_iter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ThrustRTCSharp;

namespace test_transform_iter
{
    class test_transform_iter
    {
        static void Main(string[] args)
        {
            DVVector dvalues = new DVVector(new float[] { 1.0f, 4.0f, 9.0f, 16.0f });
            Functor square_root = new Functor(new string[] { "x" }, "        return sqrtf(x);\n");
            DVTransform src = new DVTransform(dvalues, "float", square_root);
            DVVector dst = new DVVector("float", 4);
            TRTC.Copy(src, dst);
            print_array((float[])dst.to_host());

        }

        static void print_array<T>(T[] arr)
        {
            foreach (var item in arr)
            {
                Console.Write(item.ToString());
                Console.Write(" ");
            }
            Console.WriteLine("");
        }
    }
}
